src/providers/twitch/TwitchAccount.cpp

```diff
@@ -405,11 +377,19 @@ void TwitchAccount::loadUserstateEmotes()
                               [](const TwitchEmote &l, const TwitchEmote &r) {
                                   return l.name.string < r.name.string;
                               });
+
                     emoteData->emoteSets.emplace_back(newUserEmoteSet);
                 }
             },
             [] {
                 // fetching emotes failed, ivr API might be down
+            },
+            [=] {
+                // XXX(zneix): We check if this is the last iteration and if so, call the callback
```

Comment:
This might be "accurate enough" but if we really want to call the callback once all requests are done, we'll need to do some sort of syncgroup outside of the requests (see https://en.cppreference.com/w/cpp/thread/latch)